cli/internal/runsummary/run_summary.go

```diff
@@ -16,7 +25,19 @@ type RunSummary struct {
 	TurboVersion      string             `json:"turboVersion"`
 	GlobalHashSummary *GlobalHashSummary `json:"globalHashSummary"`
 	Packages          []string           `json:"packages"`
-	Tasks             []TaskSummary      `json:"tasks"`
+	Tasks             *[]*TaskSummary    `json:"tasks"`
+}
+
+// Save saves the run summary to a file
+func (summary *RunSummary) Save(singlePackage bool) error {
+	json, err := summary.FormatJSON(singlePackage)
+	if err != nil {
+		return err
+	}
+
+	nonce := "abcdef"
```

Comment:
i'm thinking we need something determinstic here, so you can run 5 run summaries and (1) identify them to all be at the same state, but (2) not overwrite each other. the timestamp lets you not overwrite, but then if you generate these files across commits/branches, you don't know which is which. 

We could also include this in the file contents itself, but I think we need a non-git solution also / something that comes from turbo itself.

Reply:
There are two types things we care about:
- linear (e.g. not-Git)
- non-linear (switch branch)

Linear will "just work" and we don't care. Non-linear is really "facts of the run" type situation. It has to be paired with the _local_ repo for it to be meaningful in all scenarios. Yeah, it makes it easier to dig out older things if the information is in the file name, but git history is a messy thing. (rebase, amend)

Reply:
>  It has to be paired with the local repo for it to be meaningful in all scenarios

I think you're optimizing for the case where thees files are shared between teammates. Is that right?

I'm imagining the scenario where you're debugging locally and you want to run turbo many times and compare. In this implementation, that would be pretty tedious, because you have to generate the summary, rename it to something, then run again. If you miss that step, you have to guess based on the timestamp. Having debugged recently with `--profile` (which similarly does not have anything helpful in the filename), I'd really like to help the developer out for this case.

Reply:
That said, @gaspar09/@mknichel want to include repo/git metadata into the run summary anyway, so that could cover some of my use case. Let's talk more about this internally

Reply:
Not even just shared! After a rebase or a commit if we had included anything from `git` in the file name it's ~meaningless unless somebody is a regular user of reflogs. Further, since we're not building from HEAD, we're also including any uncommitted content so having the file name include, say, the previous SHA, is at least misleading (in my most–charitable interpretation).

The UX for what you're describing sounds like you want a `--output-file-prefix` option which you could consistently press ⬆️ in your shell to rerun the same command and have it spit out a reasonable file name. Since it doesn't make sense to run `turbo` over and over _without_ making changes to files I'm pretty strongly opposed to grouping by git-anything. Either you're excluding the state of the index compared to the file system, you made a commit in-between and your proposed auto-prefix is nonsense, or `git checkout -b renamed-branch` (same SHA) and it's no longer relevant.

The only one I'd consider possibly viable as, say, a default value for `--output-filename-prefix` is `git symbolic-ref --short HEAD` ... which is a lot of fun because lots of folks have branches that look like `nathanhammond/feature-name` and now that's a huge pain of a filename.

Reply:
> Since it doesn't make sense to run turbo over and over without making changes to files

This is exactly the case I want to capture (example debugging use case: https://github.com/vercel/turbo/issues/4034)

1. Run turbo 5 times (without making any changes) and get 5 run summaries.
2. Make a change
3. Run turbo 5 times again and get 5 more run summaries

I want it to be obvious which set of 5 is which.

> The UX for what you're describing sounds like you want a `--output-file-prefix`

That works, but I'm thinking `TURBO_RUN_SUMMARY_DIR` as an env var instead, since we want that for another use case too)

> so having the file name include [...] the previous SHA, is [...] misleading
> I'm pretty strongly opposed to grouping by git-anything

agreed, I wasn't proposing grouping by git formally, just that if I were debugging I could _probably_ scrape by with that information (in lieu of filenames)